go/merkletree2/encoders.go

```diff
@@ -25,33 +26,68 @@ const (
 func (e EncodingType) GetEncoder() Encoder {
 	switch e {
 	case EncodingTypeBlindedSHA512_256v1:
-		return BlindedSHA512_256v1Encoder{}
+		return NewBlindedSHA512_256v1Encoder()
 	case EncodingTypeSHA512_256ForTesting:
 		return SHA512_256Encoder{}
 	default:
 		panic("Invalid EncodingType")
 	}
 }
 
-type BlindedSHA512_256v1Encoder struct{}
+// BlindedSHA512_256v1Encoder is not safe for concurrent use.
+type BlindedSHA512_256v1Encoder struct {
+	enc *codec.Encoder
+	dec *codec.Decoder
+}
 
 var _ Encoder = BlindedSHA512_256v1Encoder{}
 
-func (e BlindedSHA512_256v1Encoder) EncodeAndHashGeneric(o interface{}) (Hash, error) {
-	enc, err := msgpack.EncodeCanonical(o)
+func NewBlindedSHA512_256v1Encoder() BlindedSHA512_256v1Encoder {
+	var mh codec.MsgpackHandle
```

Comment:
can't use the same codec Handle?

Reply:
I am not sure if that object is safe for concurrent use and if each encoder needs its own copy of the handle, so I decided to err on the side of caution. If you are convinced this is not the case, I can let both the encoder and decoder share the same object, or even declare it as a package level (constant) variable